docs/DeveloperGuide.md

```diff
@@ -1,34 +1,509 @@
+
+### Life cycle of PlanNUS
+
+The <<sequence-diagram,*sequence diagram*>> below shows how different packages and classes interact with each other throughout the whole <<life-cycle,life cycle>> of PlanNUS.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/Packages_Interaction.png" alt="Sequence diagram for lifecycle of PlanNUS"/>
+</div>
+
+### Details
+
+#### Global Component
+Classes used by multiple components part of the global component of PlanNUS. This includes classes such as `App`,`Command` and `LoggingTool`. The main object classes `PartialModule`, `FullModule` and `Person` are also within the global component.
+
+**API** : `src.main.java.global`
+
+#### Storage Component
+The `Storage` component is responsible for the loading and saving of information from text files.
+
+**API** : `src.main.java.seedu.duke.storage`
+
+#### Parser Component
+For the architecture of PlanNUS, the `Parser` classes will belong under the application they will be parsing for. The role 
+of these parsers is to process the user's input and return the appropriate command with required parameters to initialise the command. The newly created objects will then be returned to the main command to be executed and thereafter, terminated. 
+
+**API** : `src.main.java.seedu.duke.parser.AppParser`, `src.main.java.seedu.duke.apps.academicplanner.AcademicPlannerParser`, `src.main.java.seedu.duke.apps.capcalculator.CapCalculatorParser` and `src.main.java.seedu.duke.apps.capcalculator.SetSuParser`
+
+#### Ui Component
+In PlanNUS, the `Ui` component is integral in initialising a `Scanner` class and passing it to methods where they require them. `Ui` also provides functions to output formatted lines to console to improve readability for the user.
+
+**API** : `src.main.java.seedu.duke.ui.Ui`
+
+
+## Implementation
+
+### Academic Calendar Planner : Add Module feature
+
+#### Current implementation
+
+Add module command is executed by `AcademicPlannerParser`. It allows users to add modules into their Academic Planner by instantiating a new `PartialModule` object and adding it into the `userModuleList` and `userModuleMap`. Both the list and hashmap are the _java API_, which are used by importing them. The `Person` object is used to encapsulate both `userModuleList` and `userModuleMap`.
+
+Additionally, the add module command extends the `Command` class and overrides its `execute()` command. An external class, `ModuleValidator` is called upon to validate the various parameters that the user has entered, as to only allow valid modules to be added to the user.
+
+Given below is an example usage scenario and how add module command behaves at each step.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_initialState.png" alt="Initial state diagram for AddModuleCommand"/>
```

Comment:
There is arrows going into the Ui  person and ModuleLoader, unclear what they are doing. The AddModuleCommand recieves some information from these 3, the diagram doesn't show what the relationship is.